### PR TITLE
GH-9: Add common pattern tests

### LINKED ISSUE #9: Write common pattern tests

### PR DESCRIPTION
**Pull request by qexat**
This PR aims to add tests to Crossandra using the framework Pytest to fix #9.

Objectives:
- [x] Write tests for common patterns
- [ ] ~~Write tests for the rest as well~~ (moved to another PR)

### CODE REVIEW

File: src/crossandra/lib.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def generate_tree(inp: Iterable[tuple[str, Enum]]) -> Tree:
             assert isinstance(_curr, dict)
             curr = _curr
         k = k[-1]
-        if (dct := curr.get(k)):
+        if dct := curr.get(k):
```
Comment (qexat) [inline src/crossandra/lib.py:40]: Accidentally ran `black` over this file by saving it :laughing:

**Comment by qexat** [comment]
Alright, I think common patterns tests are pretty much done now.

**Comment by qexat** [comment]
Given the history, I suggest to squash this PR :laughing:

File: tests/common_test.py [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+from crossandra.common import *
+from crossandra.rule import *
+
+
+# ESC = ESCAPED
+# UC = UPPERCASE
+# LC = LOWERCASE
+# POS = POSITIVE
+# NEG = NEGATIVE
+# USCORE = UNDERSCORD
```
[the remark this answers is not in the document]
Comment (qexat) [reply tests/common_test.py]: > 

lmaooo me when I'm terminally online and type "discord"

**Review by trag1c — APPROVED** [review]

**Comment by trag1c** [comment]
Big thanks! ❤️

**Pull request merged**